### PR TITLE
pulumi: fix fish shell completion

### PR DESCRIPTION
**Pull request by chenrui333**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [ ] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

relates to https://github.com/Homebrew/homebrew-core/pull/55443#issuecomment-636891673

cc @nesl247 @jaxxstorm

### CODE REVIEW

**Review by nesl247 — DISMISSED** [review]

File: Formula/pulumi.rb [hunk 1]
```diff
@@ -34,17 +35,14 @@ def install
       bin.install Dir["#{buildpath}/bin/*"]
       prefix.install_metafiles
 
-      # Install bash completion
-      output = Utils.popen_read("#{bin}/pulumi gen-completion bash")
-      (bash_completion/"pulumi").write output
+      # Install shell completions
+      system "#{bin}/pulumi gen-completion bash > pulumi.bash"
+      system "#{bin}/pulumi gen-completion zsh > _pulumi"
+      system "#{bin}/pulumi gen-completion fish > pulumi.fish"
```
Comment (SMillerDev) [inline Formula/pulumi.rb:41]: Why don't these work with the way all other formulae do this?

Comment (chenrui333) [reply Formula/pulumi.rb:41]: what are the other formulae? I think this is the same way I did with `diesel`

Comment (SMillerDev) [reply Formula/pulumi.rb:41]: I don't think I've ever seen a formula do bash redirects in general, as far as I know it's always the way you're removing.

File: Formula/pulumi.rb [hunk 2]
```diff
@@ -34,17 +35,10 @@ def install
...
-      output = Utils.popen_read("#{bin}/pulumi gen-completion bash")
-      (bash_completion/"pulumi").write output
-
-      # Install zsh completion
-      output = Utils.popen_read("#{bin}/pulumi gen-completion zsh")
-      (zsh_completion/"_pulumi").write output
-
-      # Install fish completion
-      output = Utils.popen_read("#{bin}/pulumi gen-completion fish")
-      (fish_completion/"_pulumi").write output
```
Comment (issyl0) [inline Formula/pulumi.rb:47]: Yeah, I think this just needs to change to `(fish_completion/"_pulumi.fish").write output`?

Comment (chenrui333) [reply Formula/pulumi.rb:47]: yeah, either way works (also bash completion is wrong, fixed in the PR)

Comment (issyl0) [reply Formula/pulumi.rb:47]: Cool. Using the existing (previous to this PR) methods sounds good to me to preserve existing convention, then.

**Review by nesl247 — APPROVED** [review]

**Pull request closed without merging**